Code/Fun&Learn/src/test/java/quiz/TestQuiz.java

```diff
@@ -0,0 +1,77 @@
+package test.java.quiz;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import main.java.quiz.DeutschlandF;
+import main.java.quiz.ParseQuestions;
+import main.java.quiz.Question;
+import main.java.quiz.Quiz;
+
+public class TestQuiz {
+	String eingabe = "Deutschland";	
+	String[] args;
```

Comment:
statt dem würd ich der Hauptklasse, wo die args gebraucht werden ein Boolena test verpassen, sodas bei einem Test gar nicht die Klasse erneut ausgeführt wird

Reply:
in der Klasse Quiz ein Attribut test reinschreiben, dass beim Test true und sonst false ist, dann kannst du um die Aufrufe von QuizMain.main(args) ein if (!test) drum bauen und ggf auch um Ausgaben oder so. Im Konstruktor sollte das Attribut dann entsprechend gesetzt werden